extensions/wikia/ImageReview/ImageCountGetter.class.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+class ImageCountGetter extends WikiaModel {
+
+	const QUESTIONABLE = 'questionable';
+	const REJECTED     = 'rejected';
+	const UNREVIEWED   = 'unreviewed';
+	const INVALID      = 'invalid';
+
+
+	public function getImageCounts( $reviewerId ) {
+		$counts = $this->mergeCounts(
+			$this->getCountOfAllUnassignedImages(),
+			$this->getCountOfThisUsersInProgressImages( $reviewerId )
+		);
+
+		return [
+			self::QUESTIONABLE => $counts[ ImageReviewStates::QUESTIONABLE ] ?? 0,
+			self::REJECTED => $counts[ ImageReviewStates::REJECTED ] ?? 0,
+			self::UNREVIEWED => $counts[ ImageReviewStates::UNREVIEWED ] ?? 0,
+			self::INVALID => $counts[ ImageReviewStates::INVALID_IMAGE ] ?? 0
+		];
+	}
+
+	/**
+	 * Get count of all images which have yet to be assigned a reviewer. This is for
+	 * both images that haven't gone through the first screening where they are marked
+	 * as questionable, rejected, or approved by the first level reviewers, and for
+	 * those images marked rejected or questionable which haven't been assigned to a staff
+	 * member to make the final call.
+	 * @return bool|mixed
+	 * @throws Exception
+	 */
+	private function getCountOfAllUnassignedImages() {
+		$db = $this->getDatawareDB();
+		$counts = ( new WikiaSQL() )
+			->SELECT()
+			->FIELD( 'state' )
+			->COUNT( '*' )->AS_( 'total' )
+			->FROM( 'image_review' )
+			->WHERE( 'top_200' )->EQUAL_TO( '0' )
+			->AND_( 'state' )->IN(
+				[
+					ImageReviewStates::QUESTIONABLE,
+					ImageReviewStates::REJECTED,
+					ImageReviewStates::UNREVIEWED,
+					ImageReviewStates::INVALID_IMAGE,
+				]
+			)
+			->GROUP_BY( 'state' )
+			->runLoop( $db, function( &$counts, $row ) {
+				$counts[$row->state] = $row->total;
+			} );
+
+		return $counts;
+	}
+
+	/**
+	 * Get counts for all images this user has in outstanding reviews. These are images
+	 * which have been assigned to the user, but which they haven't made an action on
+	 * (eg, marking the image as approved, rejected, questionable, etc). These images
+	 * are hidden from all other users since they have been assigned to this user as part
+	 * of creating the review.
+	 * @param $reviewerId
+	 * @return bool|mixed
+	 * @throws Exception
+	 */
+	private function getCountOfThisUsersInProgressImages( $reviewerId ) {
+		$db = $this->getDatawareDB();
+		$counts = ( new WikiaSQL() )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$counts". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)